pkg/proxy/topology.go

```diff
@@ -78,3 +78,37 @@ func FilterTopologyEndpoint(nodeLabels map[string]string, topologyKeys []string,
 	}
 	return filteredEndpoint
 }
+
+// FilterLocalEndpoint returns the node local endpoints based on configured
+// InternalTrafficPolicy.
+//
+// If InternalTrafficPolicy is PreferLocal, the number of all node local
+// endpoints will be counted. If it equals to zero, then return the original
+// array of Endpoints; otherwise, only return the node local endpoints.
+//
+// If InternalTrafficPolicy is Local, only return the node local endpoints.
+func FilterLocalEndpoint(internalTrafficPolicy *v1.ServiceInternalTrafficPolicyType, endpoints []Endpoint) []Endpoint {
```

Comment:
Can we add a unit test here similar to `TestFilterTopologyEndpoint`?

Reply:
done